compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Imports.ts

```diff
@@ -101,6 +106,8 @@ export class ProgramContext {
     this.code = code;
     this.reactRuntimeModule = getReactCompilerRuntimeModule(opts.target);
     this.suppressions = suppressions;
+    this.hasModuleScopeOptOut =
+      findDirectiveDisablingMemoization(program.node.directives) != null;
```

Comment:
It's much easier to reason about constructors when they are dumb data initialization, but this is moderately interesting program analysis. Maybe move this to be passed to the constructor as an argument?